### DISCUSSION TITLE
How to configure eslint and tsconfig?

### DISCUSSION THREAD

**Discussion by danielo515** [johnlindquist/kit#1107] [Q&A]
Hello.
I'm really interested in having my project be properly linted and type checked.
Currently, I'm getting lots of eslint errors because undefined globals, and I was wondering how can I configure so eslint and typescript knows about the same globals?

**Comment by johnlindquist** [comment]
You may need to "Reload Window" if you had your `~/.kenv` open in VS Code open during a time where something changed in `~/.kit` (such as an update/install) so it picks up on the types.

![CleanShot 2023-02-07 at 15 25 11](https://user-images.githubusercontent.com/36073/217380584-8b2604e8-5a91-4aca-a51e-5536682d73ea.png)

I'd have to google for how to configure the eslint rules.

**Comment by danielo515** [comment]
It was not as simple as just letting TS pick the types, eslint does not immmediately understand TS.
This config worked quite good, in case anyone is interested in a simple config:

```js
module.exports = {
	env: {
		es2021: true,
		node: true,
	},
	extends: 'xo',
	overrides: [
		{
			extends: ['xo-typescript'],
			files: ['*.ts'],
		},
	],
	plugins: ['@typescript-eslint'],
	parser: '@typescript-eslint/parser',
	parserOptions: {
		ecmaVersion: 'latest',
		sourceType: 'module',
		project: ['./tsconfig.json'],
		tsconfigRootDir: __dirname,
	},
	rules: {},
};
```
